### PR TITLE
fix: [CI-4056]: automountServiceToken default true on k8s

### PR DESCRIPTION
**Pull request by mtran7**
##### Screenshots:

image

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: src/modules/75-ci/components/PipelineStudio/BuildInfraSpecifications/BuildInfraSpecifications.tsx [hunk 1]
```diff
@@ -1486,6 +1486,11 @@ export default function BuildInfraSpecifications({ children }: React.PropsWithCh
                                       const infraType = val as K8sDirectInfraYaml['type']
                                       setFieldValue('buildInfraType', infraType)
                                       setBuildInfraType(val as K8sDirectInfraYaml['type'])
+                                      if (infraType === 'KubernetesDirect') {
```
Comment (tan-nhu) [inline src/modules/75-ci/components/PipelineStudio/BuildInfraSpecifications/BuildInfraSpecifications.tsx:1489]: This string should be in a const or enum

**Review by tan-nhu — APPROVED** [review]

**Review by vardanbansal-harness — APPROVED** [review]

**Comment by mtran7** [comment]
trigger cypress

**Comment by mtran7** [comment]
trigger cypress

**Comment by mtran7** [comment]
Unrelated  
1 failing

  1) Configure Datadog health source
       Custom HealthSource loads in edit mode:
     AssertionError: Timed out retrying after 4000ms: Expected to find element: `[name="metricName"]`, but never found it.

**Pull request merged**